distributed/worker.py

```diff
@@ -784,7 +784,8 @@ async def get_metrics(self):
             },
             executing={
                 key: now - self.tasks[key].start_time
-                for key in self.active_threads.values()
+                for key in self.active_threads.values() 
+                if key in self.tasks
```

Comment:
Ah, an interesting addition.  That makes sense.  I'm curious, how did this issue present?

Reply:
Thanks for fixing this @selshowk. I'd also be curious to know what failure prompted this addition

Reply:
I'm trying to reproduce it but am not able to.  It might have been with a different version of the test function which didn't work so never got committed.  I'll try a bit more to reproduce it.